pkgs/applications/video/mpv/default.nix

```diff
@@ -192,7 +194,19 @@ in stdenv.mkDerivation rec {
     cp TOOLS/umpv $out/bin
     wrapProgram $out/bin/umpv \
       --set MPV "$out/bin/mpv"
-  '';
+
+  '' + optionalString stdenv.isDarwin (''
+    mkdir -p $out/Applications
+    cp -r build/mpv.app $out/Applications
+    wrapProgram $out/Applications/mpv.app/Contents/MacOS/mpv \
+      --add-flags "--scripts=${concatStringsSep "," scripts}" \
```

Comment:
It would be nice to refactor this to avoid duplication. In these cases it may make sense to just do:

```
mkdir -p $out/Applications
cp -r build/mpv.app $out/Applications 
rm $out/Applications/mpv.app/Contents/MacOS/mpv
ln -s $out/bin/mpv $out/Applications/mpv.app/Contents/MacOS/mpv
```

Reply:
I tested this and mpv.app just quits whenever I try to open it, I don't know too much about macOS so I didn't manage to find any error messages related to this. The only method I know of was checking Console.app, but that doesn't show anything related.

At the very least I could move all the wrapProgram flags into a variable to avoid duplication that way.